CMakeLists.txt

```diff
@@ -143,155 +332,37 @@ add_library(lego1 SHARED
-  LEGO1/realtime/matrix.cpp
-  LEGO1/realtime/orientableroi.cpp
-  LEGO1/realtime/realtime.cpp
-  LEGO1/realtime/realtimeview.cpp
-  LEGO1/realtime/vector.cpp
-  LEGO1/tgl/d3drm/camera.cpp
-  LEGO1/tgl/d3drm/device.cpp
-  LEGO1/tgl/d3drm/group.cpp
-  LEGO1/tgl/d3drm/light.cpp
-  LEGO1/tgl/d3drm/mesh.cpp
-  LEGO1/tgl/d3drm/renderer.cpp
-  LEGO1/tgl/d3drm/texture.cpp
-  LEGO1/tgl/d3drm/unk.cpp
-  LEGO1/tgl/d3drm/view.cpp
-  LEGO1/viewmanager/viewlodlist.cpp
-  LEGO1/viewmanager/viewmanager.cpp
-  LEGO1/viewmanager/viewroi.cpp
 )
+register_lego1_target(lego1)
 
 if (MINGW)
   target_compile_definitions(lego1 PRIVATE DIRECTINPUT_VERSION=0x0500)
 endif()
 
-target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/util")
-target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/LEGO1")
-target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/LEGO1/omni/include")
-target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/LEGO1/lego/sources")
-target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/LEGO1/lego/legoomni/include")
-
-# Additional include directories
-target_include_directories(lego1 PUBLIC "${CMAKE_SOURCE_DIR}/3rdparty/vec")
-target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/3rdparty/flic")
-target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/3rdparty/smacker")
-
-if (ISLE_USE_SMARTHEAP)
-  add_library(SmartHeap::SmartHeap STATIC IMPORTED)
-  set_target_properties(SmartHeap::SmartHeap PROPERTIES
-    IMPORTED_LOCATION "${CMAKE_SOURCE_DIR}/3rdparty/smartheap/SHLW32MT.LIB"
-    INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_SOURCE_DIR}/3rdparty/smartheap"
-    INTERFACE_COMPILE_OPTIONS "/FI${CMAKE_SOURCE_DIR}/3rdparty/smartheap/SMRTHEAP.HPP")
-
-  target_link_libraries(lego1 PRIVATE SmartHeap::SmartHeap)
-endif()
-
-# Use internal DirectX 5 if required
-if (ISLE_USE_DX5)
-  target_include_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/3rdparty/dx5/inc")
-  target_link_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/3rdparty/dx5/lib")
+if (MSVC)
+  target_sources(lego1 PRIVATE LEGO1/LegoOmni.def)
+else()
+  target_sources(lego1 PRIVATE LEGO1/LegoOmni.mingw.def)
 endif()
 
-target_link_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/3rdparty/smacker")
-target_link_directories(lego1 PRIVATE "${CMAKE_SOURCE_DIR}/3rdparty/flic")
+target_include_directories(lego1 PUBLIC "${CMAKE_SOURCE_DIR}/util")
+target_include_directories(lego1 PUBLIC "${CMAKE_SOURCE_DIR}/LEGO1")
+target_include_directories(lego1 PUBLIC "${CMAKE_SOURCE_DIR}/LEGO1/omni/include")
+target_include_directories(lego1 PUBLIC "${CMAKE_SOURCE_DIR}/LEGO1/lego/sources")
+target_include_directories(lego1 PUBLIC "${CMAKE_SOURCE_DIR}/LEGO1/lego/legoomni/include")
 
 # Link libraries
-target_link_libraries(lego1 PRIVATE ddraw dsound dinput dxguid winmm d3drm smack flic)
+target_link_libraries(lego1 PRIVATE tglrl viewmanager realtime mxdirectx roi FLIC::FLIC Vec::Vec dinput dxguid Smacker::Smacker misc 3dmanager omni)
```

Comment:
Do we still need to specify FLIC and Smacker here since they are already linked into Omni?

Reply:
LEGO1/omni/include/mxflcpresenter.h includes `flic.h`. the smacker can go.